From 179ec2e67bf882c6bccb27f81db3d80f7eb9946e Mon Sep 17 00:00:00 2001
From: Jacob Lambert <jacob.lambert@amd.com>
Date: Fri, 12 Apr 2024 13:56:42 -0700
Subject: [PATCH] [Comgr] Don't add -nogpulib option for assembley action

We can omit setting -nogpulib even without a -rocm-path=. option
when calling the assembly action. This avoids the following warning:

warning: argument unused during compilation: '-nogpulib'
Change-Id: I66d512befbafd9382f050c45a0d3950985e8ae38
---
 amd/comgr/src/comgr-compiler.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/comgr-compiler.cpp b/src/comgr-compiler.cpp
index 143ab4e4f3db..21b233fa94b2 100644
--- a/src/comgr-compiler.cpp
+++ b/src/comgr-compiler.cpp
@@ -1758,6 +1758,9 @@ amd_comgr_status_t AMDGPUCompiler::assembleToRelocatable() {
   Args.push_back("-x");
   Args.push_back("assembler");
 
+  // -nogpulib option not needed for assembling to relocatable
+  NoGpuLib = false;
+
   return processFiles(AMD_COMGR_DATA_KIND_RELOCATABLE, ".o");
 }
 
-- 
2.44.0

